src/H5TSatomic.c

```diff
@@ -0,0 +1,403 @@
+/* * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * *
+ * Copyright by The HDF Group.                                               *
+ * All rights reserved.                                                      *
+ *                                                                           *
+ * This file is part of HDF5.  The full HDF5 copyright notice, including     *
+ * terms governing use, modification, and redistribution, is contained in    *
+ * the COPYING file, which can be found at the root of the source code       *
+ * distribution tree, or in https://www.hdfgroup.org/licenses.               *
+ * If you do not have access to either file, you may request a copy from     *
+ * help@hdfgroup.org.                                                        *
+ * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
+
+/*
+ * Purpose: C11 atomicr emulation outines
+ *
+ * Note:  Because this threadsafety framework operates outside the library,
+ *        it does not use the error stack (although it does use error macros
+ *        that don't push errors on a stack) and only uses the "namecheck only"
+ *        FUNC_ENTER_* / FUNC_LEAVE_* macros.
+ */
+
+/****************/
+/* Module Setup */
+/****************/
+
+#include "H5TSmodule.h" /* This source code file is part of the H5TS module */
+
+/***********/
+/* Headers */
+/***********/
+#include "H5private.h"  /* Generic Functions                   */
+#include "H5Eprivate.h" /* Error handling                      */
+#include "H5TSpkg.h"    /* Threadsafety                        */
+
+#ifdef H5_HAVE_THREADS
+
+#ifndef H5_HAVE_STDATOMIC_H
+
+/****************/
+/* Local Macros */
+/****************/
+
+/******************/
+/* Local Typedefs */
+/******************/
+
+/********************/
+/* Local Prototypes */
+/********************/
+
+/*********************/
+/* Package Variables */
+/*********************/
+
+/*****************************/
+/* Library Private Variables */
+/*****************************/
+
+/*******************/
+/* Local Variables */
+/*******************/
+
+/*--------------------------------------------------------------------------
+ * Function:    H5TS_atomic_init_int
+ *
+ * Purpose:     Initializes an atomic 'int' variable object with a value.
+ *
+ * Note:        Per the C11 standard, this function is not atomic and
+ *              concurrent execution from multiple threads is a data race.
+ *
+ * Return:      None
+ *
+ *--------------------------------------------------------------------------
+ */
+void
+H5TS_atomic_init_int(H5TS_atomic_int_t *obj, int desired)
+{
+    FUNC_ENTER_NOAPI_NAMECHECK_ONLY
+
+        /* Initialize mutex that protects the "atomic" value */
```

Comment:
Is there a reason the first lines of these functions are indented this way?

Reply:
> Is there a reason the first lines of these functions are indented this way?

Yes - clang-format didn't know about the "NAMECHECK" macros.  Fixed now.

Reply:
Indentation still appears to be messed up